include/glow/Backends/TraceEvents.h

```diff
@@ -149,6 +149,9 @@ class TraceContext {
   /// Sets the level of verbosity for TraceEvents.
   void setTraceLevel(TraceLevel level) { traceLevel_ = level; }
 
+  /// Get a const reference to traceEvents_ for reading, doesn't bother locking.
```

Comment:
It's the second case. I don't really care if const or not but I'm not modifying it here which is why I took it as const. We could add a const and a non-const getTraceEvents. Likewise we could lock or not. Imo the best thing to do would be to make the lock mutable and then have a const and non const one of these that both take the lock since that's difficult to abuse in the future.

Reply:
What do you mean make the lock mutable? 

Another option is we can just move the events out under the lock and process them destructively